antidox/wikiwatcher.py

```diff
@@ -4,21 +4,63 @@
 from __future__ import division
 from __future__ import print_function
 
-import argparse
 import json
-
+import pprint
+import argparse
+import requests
 import sseclient
+from googleapiclient import errors as google_api_errors
+import clean
+import perspective
 
 
+# pylint: disable=fixme, too-many-locals
 def log_event(change):
   """Logs event by printing.
 
   Args:
     change: a json object with the wikimedia change record.
   """
-  print(
-      u'user:{user} namespace:{namespace} bot:{bot} comment:{comment} title:{title}'
-      .format(**change))
+  # print(
+  #     u'user:{user} namespace:{namespace} bot:{bot} comment:{comment} title:{title}'
+  #     .format(**change))
+  print('\n########## change:')
+  from_id = (str(change['revision']['old']))
+  to_id = (str(change['revision']['new']))
+  page = ("https://en.wikipedia.org/w/api.php?action=compare&fromrev="
+          + from_id + "&torev=" + to_id + "&format=json")
+  get_page = requests.get(page)
+  response = json.loads(get_page.content)
+  revision = response['compare']['*']
+
+  text = clean.content_clean(revision).split("\n")
+
+  apikey_data, toxicity, dlp = perspective.get_client()
```

Comment:
This is creating a new Perspective and DLP client object for each detected change on Wikipedia. This works, but this is less efficient than it could be (we can reuse the same client objects instead of recreating them each time). It would be better to refactor this so that we only do `perspective.get_client()` once in this program (probably in the `watcher` function) and reuse the same client objects.

Daniel or I can help you do this if you're not sure.

Reply:
yeah, i'll need help